cmd/telemetrygen/internal/metrics/worker.go

```diff
@@ -0,0 +1,71 @@
+// Copyright The OpenTelemetry Authors
+// Copyright (c) 2018 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics // import "github.com/open-telemetry/opentelemetry-collector-contrib/cmd/telemetrygen/internal/metrics"
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric/global"
+	"go.uber.org/atomic"
+	"go.uber.org/zap"
+	"golang.org/x/time/rate"
+)
+
+type worker struct {
+	running        *atomic.Bool    // pointer to shared flag that indicates it's time to stop the test
+	numMetrics     int             // how many metrics the worker has to generate (only when duration==0)
+	totalDuration  time.Duration   // how long to run the test for (overrides `numMetrics`)
+	limitPerSecond rate.Limit      // how many metrics per second to generate
+	wg             *sync.WaitGroup // notify when done
+	logger         *zap.Logger
+}
+
+func (w worker) simulateMetrics() {
+	limiter := rate.NewLimiter(w.limitPerSecond, 1)
+	var i int
+	meter := global.Meter("telemetrygen")
+
+	index := 0
+	max := 1000
+	if w.limitPerSecond != rate.Inf {
+		max = int(w.limitPerSecond)
+	}
+
+	for w.running.Load() {
+		gauge, _ := meter.Int64UpDownCounter(fmt.Sprintf("iteration%d", index))
+		gauge.Add(context.Background(), int64(i), attribute.String("host.name", "myhost"))
```

Comment:
I hope you can bear with me being a bit nitpicky about this, let me expand on the vision I had for this (we can discuss if that vision makes sense or not in the first place).

The main point I want to defend is that I think the default behavior should be 'as simple as possible', and in future PRs we can add flags/subcommands to make the metric reported more complicated. 

In practical terms this means I think by default (ordered by how strongly I feel about them):
1. We should have a single unit-less instrument (so, we should not create a new instrument on every iteration)
2. The OTLP type should be as simple as possible (so OTLP gauge, here we are creating an OTLP non-monotonic sum)
3. There should be no attributes by default on the points (so, I would remove the `host.name` attribute)
4. The metric should have boring predictable values (I had thought about constant values, I guess linearly increasing also works?)

So my suggestion would be to do something more similar to this:

```go
gauge, _ := meter.Int64ObservableGauge("telemetrygen.example.metric")
_, err = meter.RegisterCallback(func(_ context.Context, o api.Observer) error {
    o.ObserveInt64(gauge, 1)
    return nil
}, gauge)
```

(and I think we don't really need a loop then, the metric generation is handled at the SDK level instead by calling the reader more/less frequently)

This is quite different from what you had written, so before considering the above I am curious about what your 'vision' is

Reply:
1. agreed, let me change that.
2. Let's talk tomorrow during the metting. I didn't use a gauge because its behavior is async and there is aggregation happening automatically, which defeats the purpose of creating datapoints. I don't know much so it's possible I'm off base - it's even pretty clear I'm off base. Let me try your code if I get the time.
3. ok!
4. linearly increasing really helps testers see if something is off imo.

Reply:
For (2), isn't aggregation also a problem for any other instrument, including `Counter` or `UpDownCounter`? I don't see how the instrument being async makes this more of a problem (if anything, I would expect it to be less problematic since the metric value is fetched once per collection period)